bazel_tools/pom_file.bzl

```diff
@@ -0,0 +1,221 @@
+# Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# These rules are similar to the rules in bazel_common.
+# However, our requirements are different in a few ways so we have our own version for now:
+# 1. We only include immediate dependencies not all transitive dependencies.
+# 2. We support Scala.
+# 3. We produce full pom files instead of only the dependency section.
+# 4. We have some special options to deal with our specific setup.
+
+MavenInfo = provider(
+    fields = {
+        "maven_coordinates": """
+        The Maven coordinates of this target or the one wrapped by it.
+        """,
+        "maven_dependencies": """
+        The Maven coordinates of the direct dependencies of this target.
+        """,
+    },
+)
+
+_EMPTY_MAVEN_INFO = MavenInfo(
+    maven_coordinates = None,
+    maven_dependencies = depset(),
+)
+
+_MAVEN_COORDINATES_PREFIX = "maven_coordinates="
+
+_SCALA_VERSION = "2.12"
+
+def _maven_coordinates(targets):
+    return [target[MavenInfo].maven_coordinates for target in targets if MavenInfo in target and target[MavenInfo].maven_coordinates]
+
+def jar_version(name):
+    return name.rsplit("-", 1)[1].rsplit(".", 1)[0]
+
+def _collect_maven_info_impl(_target, ctx):
+    tags = getattr(ctx.rule.attr, "tags", [])
+    deps = getattr(ctx.rule.attr, "deps", [])
+    exports = getattr(ctx.rule.attr, "exports", [])
+    jars = getattr(ctx.rule.attr, "jars", [])
+
+    # We need to detect targets generated by bazel-deps to avoid propagating transitive dependencies
+    # in that case. This is slightly tricky as bazel-deps generates a java_import that has the tag we care about
+    # and then a java_library that has the java_import in exports but also the immediate dependencies.
+    # We detect these targets by looking for a jar:jar export.
+    bazel_deps_reexport = []
+    if ctx.rule.kind == "java_import" or ctx.rule.kind == "java_library":
+        for e in exports:
+            if e.label.name == "jar" and e.label.package == "jar":
+                if MavenInfo not in e:
+                    fail("Expected maven info for jar dependency: {}".format(e.label))
+                return [e[MavenInfo]]
+    elif ctx.rule.kind == "scala_import":
+        if len(jars) != 1:
+            fail("Expected exactly one jar in a scala_import")
+        jar = jars[0]
+
+        # The following four lines corresponds to the replacements section in dependencies.yaml.
+        if jar.label.workspace_name == "io_bazel_rules_scala_scala_compiler":
+            return [MavenInfo(
+                maven_coordinates = "org.scala-lang:scala-compiler_{}:{}".format(_SCALA_VERSION, jar_version(jar.label.name)),
+                maven_dependencies = [],
+            )]
+        if jar.label.workspace_name == "io_bazel_rules_scala_scala_library":
```

Comment:
~could these be `elif`s?~

Actually, maybe this could be shortened a bit using a `dict` of workspace names mapping to maven coordinates.

Reply:
Good idea, changed!

Reply:
Thanks!